### DISCUSSION TITLE
网络封面下载改善、本地封面关联改善需求

### DISCUSSION THREAD

**Discussion by TaicEart** [zhongyang219/MusicPlayer2#440] [General]
### 网络
听本地音乐的人和在线音乐的人，对音乐文件态度是非常不同的。可以说倾向保存到本地的人，不太接受除自己以外的意志修改保存好的文件。
这需求封面也像歌词那样，可以有保存到指定文件夹的选项，因为有时欲读取的音乐文件夹是只读的——不如说习惯设为只读，更符合松鼠党（囤积癖）

顺带，发现有些人需求与我上述的类似：

- https://github.com/zhongyang219/MusicPlayer2/discussions/427
- https://github.com/zhongyang219/MusicPlayer2/issues/415

### 本地
目前MusicPlayer2（V2.75）的“外部图片文件名”操作异常麻烦。
比如许多同人音声的封面，几乎每个同人社团都有各自的命名风格，甚至有些单独放在另一文件夹“イラスト”里。
一些已知规律（.pic指代各种格式图片）：

- <标题>イラスト.pic
- ..\イラスト\\*.pic
- \*ジャケット*.pic
- \イラストデータ\\*.pic
- <RJ号>_img_main.pic
- \*ロゴ{有り|なし}\*.pic
- ……
- 这个可以列出一串。

这需求“外部图片文件名”至少允许通配符和相对路径。

**Comment by lrisora** [comment]
这里应该是允许相对路径的，请不要自行添加图片格式后缀，程序会自行在后面添加`.*`进行匹配
注意此处通配符不是程序解析的是直接递交 windows API _wfindfirst 进行文件查找，故仅支持windows路径通配符语法
另外支持的图片格式有限，有些当前格式无法识别
支持 type == L".jpg" || type == L".jpeg" || type == L".png" || type == L".gif" || type == L".bmp"
![圖片](https://user-images.githubusercontent.com/43085437/193394070-3dbf6d74-3aff-4dfd-a0f5-b32ef4494bfc.png)
